pkg/ccl/changefeedccl/changefeedbase/options.go

```diff
@@ -248,9 +248,11 @@ const (
 	SinkParamConfluentAPIKey    = `api_key`
 	SinkParamConfluentAPISecret = `api_secret`
 
-	SinkSchemeAzureKafka        = `azure-event-hub`
-	SinkParamAzureAccessKeyName = `shared_access_key_name`
-	SinkParamAzureAccessKey     = `shared_access_key`
+	SinkSchemeAzureKafka             = `azure-event-hub`
+	SinkParamAzureAccessKeyName      = `shared_access_key_name`
+	SinkParamAzureAccessKeyNameCamel = `SharedAccessKeyName`
+	SinkParamAzureAccessKey          = `shared_access_key`
+	SinkParamAzureAccessKeyCamel     = `SharedAccessKey`
```

Comment:
can you add to this map to ensure only one of each pair is set?

https://github.com/massimo-ua/cockroach/blob/e6f83cee22938d3750a929b17a7ea190b30bedaf/pkg/ccl/changefeedccl/changefeedbase/options.go#L568

Reply:
.. and then add a case to TestChangefeedErrors for that

Reply:
Hey @asg0451, apologies for the late response. From what I can tell, the way params are handled in the `buildAzureKafkaConfig` causes the params validation to not work as intended (since the params are consumed and formatted into the params string [here](https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/changefeedccl/sink_kafka.go#L1081)).
I've implemented a workaround for this, but I'm not certain if it's the best approach.

Reply:
but doesnt the `incompatibleOptionsMap` validation happen way before we get to that point?

Reply:
 That's I was trying to say. Adding  SinkParamAzureAccessKey and SinkParamAzureAccessKeyCamel has no effect. Execution continues up to the moment when it fails because `nope` address doesn't exist

Reply:
I believe this occurs because these parameters are part of the URL query string rather than standard options, so they are not validated by the options validator and must be processed separately. See the example below
```
CREATE CHANGEFEED FOR foo INTO azure-event-hub://nope:9999?SharedAccessKey=redacted&shared_access_key=redacted&shared_access_key_name=saspolicyhistory
```

Reply:
AH yes of course. You're right.